core/runner/src/test/java/com/cognifide/aet/runner/distribution/dispatch/CollectorJobSchedulerTest.java

```diff
@@ -95,17 +95,19 @@ public void add_expectMessageSent() throws Exception {
     verify(producer, times(1)).send(Matchers.<Destination>any(), Matchers.<Message>any());
   }
 
-
-  @Ignore("Ignored for some unexpected test failure on Linux")
-  @Test(timeout = 2000l, expected = IllegalStateException.class)
-  public void add_whenAddedMessageExists_expectIllegalStateException() throws Exception {
+  @Test
+  public void add_messageQueueIsConsumedBySafeRunMethod() throws Exception {
     String correlationID = "98765432100";
+    Queue<MessageWithDestination> messagesQueue1 = mockMessagesQueue(Mockito.mock(Message.class),
+            Mockito.mock(Destination.class), correlationID);
+    Queue<MessageWithDestination> messagesQueue2 = mockMessagesQueue(Mockito.mock(Message.class),
+            Mockito.mock(Destination.class), correlationID);
 
-    Queue<MessageWithDestination> messagesQueue1 = mockMessagesQueue(Mockito.mock(Message.class), Mockito.mock
-            (Destination.class), correlationID);
-    Queue<MessageWithDestination> messagesQueue2 = mockMessagesQueue(Mockito.mock(Message.class), Mockito.mock
-            (Destination.class), correlationID);
     tested.add(messagesQueue1, correlationID);
+    // messageQueues will be processed and removed from messagesMap
+    // in safeRun method of CollectorJobScheduler
+    // so it is possible to add another with the same correlationId
+    Thread.sleep(1000);
```

Comment:
Would this be possible to make here some active waiting? So that when we run this test on a slow machine this test will not fail (now it may happen that this 1 second is not enough).
In other way pleas remove this test.